src/rest_framework_dso/fields.py

```diff
@@ -124,9 +130,39 @@ def get_related_list_ids(self, instances) -> list:
             related_mgr = getattr(instance, self.attname, None)
             if related_mgr is None:
                 continue
-            ids |= set(r.pk for r in related_mgr.all())
+            source_is_temporal = self.parent_model.is_temporal()
+            target_is_temporal = related_mgr.model.is_temporal()
+            if not source_is_temporal and target_is_temporal:
+
+                ids |= set(self._get_temporal_ids(instance, related_mgr))
+            else:
+                ids |= set(r.pk for r in related_mgr.all())
         return list(ids)
 
+    def _get_temporal_ids(self, instance, related_mgr):
+        (
+            identificatie_fieldname,
+            volgnummer_fieldname,
+        ) = related_mgr.model._table_schema.identifier
+        source_field_name = related_mgr.source_field_name
+        source_id = instance.id
+        tussentabel_filter_params = {source_field_name: source_id}
+        target_id_field = f"{related_mgr.target_field_name}_id"
+        tussentabel_items = [
+            getattr(item, target_id_field, None)
+            for item in related_mgr.through.objects.filter(**tussentabel_filter_params)
+        ]  # de items uit de tussen tabel.
+        target_filter_params = {f"{identificatie_fieldname}__in": tussentabel_items}
+        order_param = f"-{volgnummer_fieldname}"
+        ids = [
+            item.pk
+            for item in related_mgr.model.objects.filter(
+                **target_filter_params
+            ).order_by(order_param)[:1]
+        ]  # de items uit de buurten tabel.
```

Comment:
```suggestion
        ids = [
            related_mgr.model.objects.filter(
                **target_filter_params
            ).order_by(order_param)
            .values_list('pk', flat=True)[:1]
        ]  # de items uit de buurten tabel.
```

Hiermee haal je alleen de ID's op. Huidige aanpak is inefficient: dan hoe je een `SELECT *` op de tabel, construeer je Django models voor alle resultaten, om deze vervolgens weer weg te gooien.

Reply:
Dank je! Op zich haalt hij maar 1 element op (de [:1] vertaalt in een LIMIT), maar dit scheelt idd wat overhead.